### PR TITLE
Support project code

### PR DESCRIPTION
**Pull request by auricgoldfinger**
### Checklist

- [x] I've run `./gradlew test` from the root directory to see all new and existing tests pass _(No new tests are failing :wink: )_
- [x] I've followed the _harvest-client_ code style
- [x] I've read the [Contribution Guidelines][contribution guidelines]
- [x] I've updated the documentation if necessary.

### Motivation and Context

In the Harvest API, there's a `code` field which is accessible to non-admin users. We're using this code to identify some projects. This field is not accessible in `ImmutableProjectReferenceDto`

### Description

I added the `code` field which is available in hte [Project Assignments](https://help.getharvest.com/api-v2/users-api/users/project-assignments/) response

[contribution guidelines]: ../CONTRIBUTING.md

### CODE REVIEW

**Comment by mnembrini** [comment]
First of all thanks for the PR! 
I will be able to take a look at this on Friday the latest, maybe before that if my schedule frees up.

**Comment by mnembrini** [comment]
How did you run the tests? When I check out this branch they fail because the code seems to be an optional field
See 
image

Unfortunately we are not setup to run CI tests from forks, but I can just run them locally.

File: src/main/java/ch/aaap/harvestclient/domain/reference/dto/ProjectReferenceDto.java [hunk 1]
```diff
@@ -10,4 +10,6 @@
 @Value.Immutable
 public interface ProjectReferenceDto extends BaseReferenceDto, Reference<Project> {
 
+   String getCode();
```
Comment (mnembrini) [inline src/main/java/ch/aaap/harvestclient/domain/reference/dto/ProjectReferenceDto.java:13]:
```suggestion
@Nullable
   String getCode();
```
This code is optional

**Review by mnembrini — CHANGES_REQUESTED** [review]
if you make it nullable the tests are green

**Comment by auricgoldfinger** [comment]
I had troubles with the tests indeed! Some of them failed on master and the same failed on my branch, hence I assumed everything was ok. Problem is probably that I don't have access to a Harvest admin account...

**Comment by mnembrini** [comment]
Yes unfortunately there is no test API from Harvest and no easy programmable way to create a test account, so the tests just need you to manually create 2 harvest trial accounts. These last only 1 month so I have to remake them every time they expire. I asked them and this method is the best they can offer 😞

**Comment by mnembrini** [comment]
@auricgoldfinger Thanks for the contribution. Just as a curiosity, where are you using harvest-client? Is it working well for you?

**Comment by auricgoldfinger** [comment]
It's working great, indeed. At the office, we use different ways for time tracking. Harvest is one of them, Jira is another. 
There are plugins in Jira that try to keep Harvest in sync, but they're rather limited and it's not working very well for me: I have to be very careful or they log "0 h" in the Jira worklog.

On top of that, I keep track of every start- and stop moment for verification. Both Harvest an Jira are configured as "timer" and don't take start- and stop-times into account. So I have to do that manually. This means that I have to log my time in three different systems, and that's quite some overhead.

So I wrote a tool with which I can 
* Start and stop time tracking with custom description and Harvest Task type. In Harvest it starts the timer and I keep track of the start- and stop times outside of Harvest.
* Optionally link a Jira issue (Harvest external link)
* Sync the tracked time of Harvest back into Jira when the Harvest time tracking is stopped.

This way, I saved myself a lot of manual work. And your library saved me from writing the Harvest API connection myself :wink:

**Pull request merged**